generate_files.sh

```diff
@@ -1,26 +1,53 @@
-# RUNFILES_DIR variable set.
-
-# --- begin runfiles.bash initialization v2 ---
-# Copy-pasted from the Bazel Bash runfiles library v2.
-set -uo pipefail; f=bazel_tools/tools/bash/runfiles/runfiles.bash
-source "${RUNFILES_DIR:-/dev/null}/$f" 2>/dev/null || \
-  source "$(grep -sm1 "^$f " "${RUNFILES_MANIFEST_FILE:-/dev/null}" | cut -f2- -d' ')" 2>/dev/null || \
-  source "$0.runfiles/$f" 2>/dev/null || \
-  source "$(grep -sm1 "^$f " "$0.runfiles_manifest" | cut -f2- -d' ')" 2>/dev/null || \
-  source "$(grep -sm1 "^$f " "$0.exe.runfiles_manifest" | cut -f2- -d' ')" 2>/dev/null || \
-  { echo>&2 "ERROR: cannot find $f"; exit 1; }; f=; set -e
-# --- end runfiles.bash initialization v2 ---
-
-external/com_github_google_flatbuffers/flatc --ts --ts-no-import-ext -o ${BUILD_WORKSPACE_DIRECTORY}/src/test/ src/ByteVector.fbs
-external/com_github_google_flatbuffers/flatc --binary --schema -o ${BUILD_WORKSPACE_DIRECTORY}/src/test/ src/ByteVector.fbs
-
-# TODO(jkuszmaul): Also codegen the reflection typescript code (I'm currently
-# avoiding this because I want to fix some upstream flatbuffers bazel rules).
-
-# Subtle diffs in the .bfbs files will be hard to track; so long as the
-# generated typescript is stable, we are probably fine.
-cd ${BUILD_WORKSPACE_DIRECTORY}
-git diff --exit-code -- ':!*.bfbs' ':!*lock*'
+set -eo pipefail
+
+flatc_version="2.0.8"
+expected_flatc_version_output="flatc version $flatc_version"
+
+# Detect whether an installed flatc is usable or download it if requested.
+flatc_executable=$(command -v flatc || echo "")
+if [[ -n "$flatc_executable" ]] && [[ $(flatc --version) = "$expected_flatc_version_output" ]]; then
+  echo "Using preinstalled flatc $flatc_version from \$PATH."
+elif [[ -n "$flatc_executable" ]] && [[ -z "$DOWNLOAD_FLATC" ]]; then
+  echo "Files have been generated with flatc $flatc_version; you have $(flatc --version)."
+  echo "Please install the correct version or set DOWNLOAD_FLATC=<platform> to download it."
+  exit 1
+else
+  case "$DOWNLOAD_FLATC" in
+    "")
+      echo "flatc $flatc_version is not installed. Install flatc or set DOWNLOAD_FLATC=<platform> to download it."
+      exit 1
+      ;;
+    linux)
+      flatc_artifact_name=Linux.flatc.binary.clang++-12.zip
+      ;;
+    mac)
+      flatc_artifact_name=Mac.flatc.binary.zip
+      ;;
+    *)
+      echo "Unsupported platform $DOWNLOAD_FLATC. Install flatc manually to continue."
+      exit 1
+  esac
+  flatc_url="https://github.com/google/flatbuffers/releases/download/v$flatc_version/$flatc_artifact_name"
+  echo "Downloading $flatc_url..."
+  mkdir -p tmp
+  curl -sL -o tmp/flatc.zip "$flatc_url"
+  unzip -o -d tmp tmp/flatc.zip
+  flatc_executable="tmp/flatc"
+fi
+
+"$flatc_executable" --version
+
+echo "Downloading reflection.fbs..."
+curl -sL -o src/reflection.fbs "https://raw.githubusercontent.com/google/flatbuffers/refs/tags/v$flatc_version/reflection/reflection.fbs"
```

Comment:
Should this get put in `gen/`?

Reply:
Technically it's not generated...? It is vendored though. I could move it into a vendor folder if you like? Also fine with putting it in gen if that's preferred.

Reply:
A separate folder would be nice. Something to make clear that it is coming from elsewhere.